### ISSUE TITLE
Resource leak during automated report generation (subscriptions)

### ISSUE THREAD

**Issue by siepkes** [metabase/metabase#50726]
### Describe the bug

At 08:00 (local time) a lot of automated reports are generated for subscriptions in our Metabase install. It seems every day Metabase leaks some resources when this happens. Until the point where Metabase can't really function anymore due to GC pressure and has to be restarted.

![Image](https://github.com/user-attachments/assets/3de70c8a-b7fb-44b1-9985-64788d64aacd)

As far as I can tell this resource leak seems to be only present for the report generation for subscriptions.

I took a quick look with Eclipse MAT which seems to imply this is related to the generation of SVG's. The following screenshot is from Eclipse MAT when Metabase has been running for a couple of days. There was no activity (no visitors, no subscription report generation, etc.) at the time the heap dump was made:

![Image](https://github.com/user-attachments/assets/e6c9c05f-d429-49f9-ba9f-fe1eda01d9f8)

![Image](https://github.com/user-attachments/assets/7372aee3-b868-441b-8d7f-face4e0b0b31)

If your interested I can send the heap dump via a private channel.

### To Reproduce

I haven't been able to create a standalone reproducer.

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:133.0) Gecko/20100101 Firefox/133.0",
    "vendor": ""
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres"
    ],
    "run-mode": "prod",
    "plan-alias": "",
    "version": {
      "date": "2024-11-15",
      "tag": "v0.50.35-SNAPSHOT",
      "hash": "cd31b54"
    },
    "settings": {
      "report-timezone": "Europe/Amsterdam"
    },
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.13"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    }
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "21.0.3-internal-adhoc.pkgsrc.jdk-il-jdk-21.0.3-ga",
    "java.vendor": "N/A",
    "java.vendor.url": "https://openjdk.org/",
    "java.version": "21.0.3-internal",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "21.0.3-internal-adhoc.pkgsrc.jdk-il-jdk-21.0.3-ga",
    "os.name": "SunOS",
    "os.version": "5.11",
    "user.language": "en",
    "user.timezone": "Europe/Amsterdam"
  }
}
```

### Severity

Medium

**Comment by paoliniluis** [comment]
@siepkes just curious: are you running Metabase on Solaris?

**Comment by paoliniluis** [comment]
@siepkes we have a PR almost ready to ship, will you be able to test it?

**Comment by siepkes** [comment]
Yes, definitely! I assume you want me to test https://github.com/metabase/metabase/pull/51685 ?

**Comment by paoliniluis** [comment]
@siepkes yes, do you know how to make a custom build?

**Comment by siepkes** [comment]
Yes, I backported the patch onto the `0.52.4` release and deployed it. It will probably take a couple of days to see if the memory usage does not gradually increase anymore.

**Comment by siepkes** [comment]
First of all, best wishes to everyone! 

Bit to soon to tell if the problem is a 100% solved but the results look good so far.

![Image](https://github.com/user-attachments/assets/d293e597-6410-47c3-af26-1ca3fbf32429)

At point 1 the application was rebooted (hence the drop in heap size). Point 2 is the first reports being generated and send at 07:00. There is a bit of an increase in heap size after Metabase is done, but that seems normal to me the first time those code paths are activated. Point 3 is the second time reports are generated at 07:00. The heap size returns to normal after it's done generating the reports.

I will keep monitoring this for a couple days.

**Comment by siepkes** [comment]
The memory usage seems to remain constant so I think by now we can say [this](https://github.com/metabase/metabase/pull/51685) fixes this issue!

![Image](https://github.com/user-attachments/assets/38d34083-a07f-448a-8285-eee06a59f48c)

**Comment by paoliniluis** [comment]
great news, closing
